explainer.md

```diff
@@ -0,0 +1,329 @@
+[UIAutomation](https://msdn.microsoft.com/en-us/library/windows/desktop/ee684009.aspx)
+on Windows, or
+[NSAccessibility](https://developer.apple.com/library/mac/documentation/AppKit/Reference/NSAccessibility_Protocol_Reference/)
+on OS X.
+These APIs allow an application to expose a tree of objects representing the application's interface,
+typically with the root node representing the application window,
+with various levels of grouping node descendants down to individual interactive elements.
+This is referred to as the **accessibility tree**.
+
+An assistive technology user interacts with the application almost exclusively via this API,
+as the assistive technology uses it both to create the alternative interface,
+and to route user interaction events triggered by the user's commands to the assistive technology.
+
+![Flow from application UI to accessibility tree to assistive technology to user](images/a11y-tree.jpg)
+
+For example, a [VoiceOver](https://www.apple.com/voiceover/info/guide/) user might press the key combination
+"Control Option Spacebar" to indicate that they wish to click the element which the screen reader is currently visiting.
+These keypresses would never be passed to the application,
+but would be interpreted by the screen reader,
+which would then call the `accessibilityPerformPress()` function on the element via the API.
+This is then routed back to the DOM as a `click` event by the browser.
+
+### Accessibility node properties
+
+Each node in the accessibility tree may be referred to as an **accessibility node**.
+An accessibility node always has a **role**, indicating its semantic purpose.
+This may be a grouping role,
+indicating that this node merely exists to contain and group other nodes,
+or it may be an interactive role,
+such as `"button"`.
+
+![Accessibility nodes in an accessibility tree, showing roles, names, states and properties](images/a11y-node.jpg)
+
+The user, via assistive technology, may explore the accessibility tree at various levels.
+They may interact with grouping nodes,
+such as a landmark element which helps a user navigate sections of the page,
+or they may interact with interactive nodes,
+such as a button.
+In both of these cases,
+the node will usually need to have a **label** (often referred to as a **name**)
+to indicate the node's purpose in context.
+For example, a button may have a label of "Ok" or "Menu".
+
+Accessibility nodes may also have other properties,
+such as the current **value** (e.g. `"10"` for a range, or `"Jane"` for a text input),
+or **state** information (e.g. `"checked"` for a checkbox, or `"focused"`).
+
+Interactive accessibility nodes may also have certain **actions** which may be performed on them.
+For example, a button may expose a `"press"` action.
+
+These properties and actions are referred to as the *semantics* of a node.
+
+## Existing accessibility APIs
+
+The web has rich support for making applications accessible,
```

Comment:
Should this have a subheading "### Web platform accessibility APIs" to match the other subheadings later on?